cluster/v1beta1/types_placement.go

```diff
@@ -94,6 +94,50 @@ type PlacementSpec struct {
 	// certain taints to be selected by placements with matching tolerations.
 	// +optional
 	Tolerations []Toleration `json:"tolerations,omitempty"`
+
+	// DecisionStrategy divide the created placement decision to groups and define number of clusters per decision group.
+	// +optional
+	DecisionStrategy DecisionStrategy `json:"decisionStrategy,omitempty"`
+}
+
+// DecisionGroup define a subset of clusters that will be added to placementDecisions with groupName label.
+type DecisionGroup struct {
+	// Group name to be added as label value to the created placement Decisions labels with label key cluster.open-cluster-management.io/decision-group-name
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^[a-zA-Z0-9][-A-Za-z0-9_.]{0,61}[a-zA-Z0-9]$"
+	// +required
+	GroupName string `json:"groupName,omitempty"`
+
+	// LabelSelector to select clusters subset by label.
+	// +kubebuilder:validation:Required
+	// +required
+	ClusterSelector ClusterSelector `json:"groupClusterSelector,omitempty"`
+}
+
+// Group the created placementDecision into decision groups based on the number of clusters per decision group.
+type GroupStrategy struct {
+	// DecisionGroups represents a list of predefined groups to put decision results.
+	// +optional
+	DecisionGroups []DecisionGroup `json:"decisionGroups,omitempty"`
+
+	// ClustersPerDecisionGroup is a specific number or percentage of the total selected clusters.
+	// The specific number will divide the placementDecisions to decisionGroups each group has max number of clusters equal to that specific number.
+	// The percentage will divide the placementDecisions to decisionGroups each group has max number of clusters based on the total num of selected clusters and percentage.
+	// ex; for a total 100 clusters selected, ClustersPerDecisionGroup equal to 20% will divide the placement decision to 5 groups each group should have 20 clusters.
+	// Default is having all clusters in a single group.
+	// If the DecisionGroups field defined, it will be considered first to create the decisionGroups then the ClustersPerDecisionGroup will be used to determine the rest of decisionGroups.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +kubebuilder:default:="100%"
+	// +optional
+	ClustersPerDecisionGroup string `json:"clustersPerDecisionGroup,omitempty"`
```

Comment:
Using `intstr.IntOrString` here?

Reply:
Using string should be fine from the perspective of functional implementation. 

One benefit of using `IntOrString` I can think is that it's in line with user habits, like k8s [deployment](RollingUpdateDeployment) `maxSurge` and `maxUnavailable`, user don't need to warp "" for an integer in yaml. 
```
The Placement "demo1" is invalid: spec.decisionStrategy.groupStrategy.clustersPerDecisionGroup: Invalid value: "integer": spec.decisionStrategy.groupStrategy.clustersPerDecisionGroup in body must be of type string: "integer"
```

For the format check, not sure if we can check it in code like https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go#LL212C6-L212C32 and report error in placement "misconfigured" condition.

Reply:
Nice, wasn't aware that intOrString type has function to validate the percentage. okay, changing it to intOrString